tests/test-class-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * Class WC_Payment_Gateway_WCPay_Test
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+/**
+ * WC_Payment_Gateway_WCPay unit tests.
+ */
+class WC_Payment_Gateway_WCPay_Subscriptions_Test extends WP_UnitTestCase {
+	const USER_ID           = 1;
+	const CUSTOMER_ID       = 'cus_mock';
+	const PAYMENT_METHOD_ID = 'pm_mock';
+	const CHARGE_ID         = 'ch_mock';
+	const PAYMENT_INTENT_ID = 'pi_mock';
+
+	/**
+	 * System under test.
+	 *
+	 * @var WC_Payment_Gateway_WCPay_Subscriptions_Compat
+	 */
+	private $wcpay_gateway;
+
+	/**
+	 * Mock WC_Payments_Customer_Service.
+	 *
+	 * @var WC_Payments_Customer_Service|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_customer_service;
+
+	/**
+	 * Mock WC_Payments_Token_Service.
+	 *
+	 * @var WC_Payments_Token_Service|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_token_service;
+
+	/**
+	 * Mock WC_Payments_API_Client.
+	 *
+	 * @var WC_Payments_API_Client|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_api_client;
+
+	/**
+	 * WC_Payments_Account instance.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $wcpay_account;
+
+	public function setUp() {
+		parent::setUp();
+
+		wp_set_current_user( self::USER_ID );
+
+		$this->mock_api_client = $this->getMockBuilder( 'WC_Payments_API_Client' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->wcpay_account = new WC_Payments_Account( $this->mock_api_client );
+
+		$this->mock_customer_service = $this->getMockBuilder( 'WC_Payments_Customer_Service' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->mock_token_service = $this->getMockBuilder( 'WC_Payments_Token_Service' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->wcpay_gateway = new \WC_Payment_Gateway_WCPay_Subscriptions_Compat(
+			$this->mock_api_client,
+			$this->wcpay_account,
+			$this->mock_customer_service,
+			$this->mock_token_service
+		);
+	}
+
+	public function test_add_token_to_order_should_add_token_to_subscriptions() {
+		$original_order = WC_Helper_Order::create_order( self::USER_ID );
+		$subscriptions  = [
+			WC_Helper_Order::create_order( self::USER_ID ),
+			WC_Helper_Order::create_order( self::USER_ID ),
+			WC_Helper_Order::create_order( self::USER_ID ),
+		];
+		$orders         = array_merge( [ $original_order ], $subscriptions );
+
+		$this->mock_wcs_get_subscriptions_for_order( $subscriptions );
+
+		$token = WC_Helper_Token::create_token( 'new_payment_method', self::USER_ID );
+
+		$this->wcpay_gateway->add_token_to_order( $original_order, $token );
+
+		foreach ( $orders as $order ) {
+			$payment_methods = $order->get_payment_tokens();
+			$this->assertNotFalse( end( $payment_methods ) );
+			$this->assertEquals( $token->get_id(), end( $payment_methods ) );
```

Comment:
Isn't the second assertion enough? Or do we need an assertion that `$token->get_id()` isn't `false`?

I feel like we can trust that the mock token we create has an ID, and thus the second assertion should be enough 🤔

Reply:
There are also similar assertions in loops further down 🙂.

Reply:
I think the second assertion is enough, if a token fails to be created the tests would fail before we get there. So it's guaranteed that we're not going to have a token with `ID = false`. I've removed them in https://github.com/Automattic/woocommerce-payments/pull/875/commits/d05da5e6cc9a653c44a98ebd5ac492ca37e3bf5b.